src/test/plugins/ucc_discovery_plugin_test.cpp

```diff
@@ -256,6 +282,65 @@ TEST_P(UccDiscoveryPluginMultiEncodingTest, ValidateCandidates) {
   EXPECT_FALSE(constraints_B.contains({{ColumnID{1}}, KeyConstraintType::UNIQUE}));
 }
 
+TEST_P(UccDiscoveryPluginMultiEncodingTest, ValidateCandidatesAfterDeletion) {
+  _encode_table(_table_A, GetParam());
+
+  // Delete row of _table_A that had a duplicate value regarding column 1 such that column 1 is unique afterwards.
+  _delete_row(_table_A, 3);
+
+  // We are only interested in column 1, since it was not unique before the deletion but should be now
+  const auto ucc_candidates = UccCandidates{{"uniquenessTestTableA", ColumnID{1}}};
+
+  _validate_ucc_candidates(ucc_candidates);
+
+  // Collect constraints known for the tables
+  const auto& constraints_A = _table_A->soft_key_constraints();
+  EXPECT_EQ(constraints_A.size(), 1);
+  EXPECT_TRUE(constraints_A.contains({{ColumnID{1}}, KeyConstraintType::UNIQUE}));
+}
+
+TEST_P(UccDiscoveryPluginMultiEncodingTest, RevalidationUpdatesValidationTimestamp) {
+  _encode_table(_table_A, GetParam());
+
+  // We are only interested in column 1, since it was not unique before the deletion but should be now
+  const auto ucc_candidates = UccCandidates{{"uniquenessTestTableA", ColumnID{0}}};
+
+  _validate_ucc_candidates(ucc_candidates);
+
+  // Perform a transaction that does not affect table A but increments the global Commit ID
+  _delete_row(_table_B, 0);
+
+  // Collect constraints known for the tables
+  const auto& constraints_A = _table_A->soft_key_constraints();
+  const auto column_1_constraint = constraints_A.find({{ColumnID{0}}, KeyConstraintType::UNIQUE});
+  EXPECT_EQ(constraints_A.size(), 1);
+  EXPECT_TRUE(constraints_A.contains({{ColumnID{0}}, KeyConstraintType::UNIQUE}));
+  const auto first_validation_timestamp = column_1_constraint->last_validated_on();
+
+  _validate_ucc_candidates(ucc_candidates);
+  EXPECT_GT(column_1_constraint->last_validated_on(), first_validation_timestamp);
+}
+
+TEST_P(UccDiscoveryPluginMultiEncodingTest, DeletionOfModifiedUCC) {
+  _encode_table(_table_A, GetParam());
+
+  // Insert unique column as candidate
+  auto ucc_candidates = UccCandidates{{"uniquenessTestTableA", ColumnID{0}}};
+
+  _validate_ucc_candidates(ucc_candidates);
+
+  // Collect constraints known for the tables
+  const auto& constraints_A = _table_A->soft_key_constraints();
+  EXPECT_TRUE(constraints_A.contains({{ColumnID{0}}, KeyConstraintType::UNIQUE}));
+
+  // Insert table data into the table again -> creates duplicate for every row
+  _duplicate_table(_table_name_A);
+
+  _validate_ucc_candidates(ucc_candidates);
+  const auto& constraints_B = _table_A->soft_key_constraints();
+  EXPECT_FALSE(constraints_B.contains({{ColumnID{0}}, KeyConstraintType::UNIQUE}));
+}
+
```

Comment:
Pick here: I'd love to see some more test cases, e.g., no CommitID ipdate for schema-given UCC, Updates (since it's delete + insert w/ and w/o duplicate), validation while transactions are active (insert/delete ops are executed, but not committed), ...